utils/rules.js

```diff
@@ -0,0 +1,28 @@
+const { rule, and } = require('graphql-shield');
+const { UserInputError } = require('apollo-server');
+const {
+  INVALID_PERMISSIONS,
+  USER_NOT_AUTHORIZED,
+  WRONG_CREDENTIALS,
+} = require('../error-messages/user.messages');
+
+const isAuthorized = rule()(async (parent, args, context, info) => {
+  if (context.user) return true;
+  return new UserInputError(USER_NOT_AUTHORIZED, { statusCode: 401 });
+});
+
+const isAdmin = rule()(async (parent, args, context, info) => {
+  if (context.user.role === 'admin') return true;
+  return new UserInputError(INVALID_PERMISSIONS, { statusCode: 403 });
+});
+
+const isTheSameUser = rule()(async (parent, args, context, info) => {
```

Comment:
Why do we need this kind of check, which operations should have this type of restriction?

Reply:
We have functions such as getUserById and updateUserById. It would be useful to use them